tests/ha/test_multi_server_pod_failure.py

```diff
@@ -629,3 +630,484 @@ def test_writes_till_kserver_pods_fail(self):
+
+    # pylint: disable=too-many-statements
+    # pylint: disable=too-many-locals
+    @pytest.mark.ha
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-40570")
+    @CTFailOn(error_handler)
+    def test_cont_ios_during_kserver_pods_down_safe(self):
+        """
+        Test to verify continuous IOs while k server pods are failing one by one using
+        scale replicas method
+        """
+        LOGGER.info("STARTED: Test to verify continuous IOs while k server pods are failing one "
+                    "by one using scale replica")
+
+        LOGGER.info("Step 1: Perform Continuous WRITEs-READs-verify during k server pods are "
+                    "going down")
+        users = self.mgnt_ops.create_account_users(nusers=1)
+        self.test_prefix = 'test-40570'
+        self.s3_clean = users
+        event = threading.Event()
+        output = Queue()
+        args = {'s3userinfo': list(users.values())[0], 'log_prefix': self.test_prefix,
+                'nclients': 1, 'nsamples': 3, 'skipcleanup': False, 'output': output}
+        thread = threading.Thread(target=self.ha_obj.event_s3_operation,
+                                  args=(event,), kwargs=args)
+        thread.daemon = True  # Daemonize thread
+        thread.start()
+        LOGGER.info("Step 1: Successfully started WRITEs-READs-verify in background")
+        time.sleep(HA_CFG["common_params"]["30sec_delay"])
+
+        LOGGER.info("Step 2: Shutdown the server pods by making replicas=0 (safe)")
+        LOGGER.info("Get server pod names to be deleted")
+        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.SERVER_POD_NAME_PREFIX)
+        self.pod_name_list = random.sample(pod_list, self.kvalue)
+        LOGGER.info("Get data pod names")
+        data_pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.POD_NAME_PREFIX)
+        LOGGER.info("Combine data and server pods names")
+        server_data_pod_list = pod_list + data_pod_list
+        for count, pod_name in enumerate(self.pod_name_list):
+            count += 1
+            pod_data = list()
+            pod_data.append(
+                self.node_master_list[0].get_pod_hostname(pod_name=pod_name))  # hostname
+            LOGGER.info("Deleting %s server pod %s", count, pod_name)
+            event.set()
+            resp = self.node_master_list[0].create_pod_replicas(num_replica=0, pod_name=pod_name)
+            LOGGER.debug("Response: %s", resp)
+            assert_utils.assert_false(resp[0], f"Failed to shutdown server pod {pod_name} "
+                                               f"by making replicas=0")
+            pod_data.append(resp[1])  # deployment_name
+            self.restore_pod = self.deploy = True
+            self.restore_method = const.RESTORE_SCALE_REPLICAS
+            self.pod_dict[pod_name] = pod_data
+            LOGGER.info("Deleted %s server pod %s by making replicas=0 "
+                        "(safe)", count, pod_name)
+            event.clear()
```

Comment:
clear event after we check that cluster is indeed has failures and some srevices are down

Reply:
For k pod down at once case, we will have to clear event here only right? To create expected scenario?